tests/test_codecs/test_codecs.py

```diff
@@ -370,17 +362,66 @@ def test_invalid_metadata(store: Store) -> None:
         )
 
 
+@pytest.mark.parametrize("store", ["local", "memory"], indirect=["store"])
+def test_invalid_metadata_create_array(store: Store) -> None:
+    spath3 = StorePath(store, "invalid_order")
+    with pytest.raises(TypeError):
+        zarr.create_array(
+            spath3,
+            shape=(16, 16),
+            chunks=(16, 16),
+            dtype=np.dtype("uint8"),
+            fill_value=0,
+            filters=[
+                TransposeCodec(order="F"),  # type: ignore[arg-type]
+            ],
+        )
+    spath5 = StorePath(store, "invalid_inner_chunk_shape")
+    with pytest.raises(ValueError):
+        zarr.create_array(
+            spath5,
+            shape=(16, 16),
+            shards=(16, 16),
+            chunks=(8,),
+            dtype=np.dtype("uint8"),
+            fill_value=0,
+        )
+    spath6 = StorePath(store, "invalid_inner_chunk_shape")
```

Comment:
it's fine to keep this here but I think this case should be handled by the sharding codec test suite